### ISSUE TITLE
Textboxes drawn over text

### ISSUE THREAD

**Issue by Opaque79** [LostArtefacts/TRX#748]
Since 2.13, the text boxes are being drawn above the text obscuring it. Any ideas?
![Tomb1Main 2023-03-07 12-27-19-11](https://user-images.githubusercontent.com/127208702/223422364-99f3335f-9979-4a94-8fb2-4f2a0c7be994.png)

The passport textures also glitch while it's opening too
https://i.imgur.com/abYCCk2.png

**Comment by aredfan** [comment]
My guess is you'll need to overwrite the gameflow files with the latest ones, see if that fixes your issue.

**Comment by Opaque79** [comment]
Unfortunately, I've already tried that and it didn't work. I will try again though.

EDIT: No joy unfortunately, even with no changes made to the cfg files

**Comment by walkawayy** [comment]
Maybe we need to try and figure out what exact hardware is having issues. I have an AMD 7700x CPU and an Nvidia 1070 GPU with no issue.

Could anyone having this problem please post their specs?

**Comment by oziphantom** [comment]
probably something to do with the drivers not following the same sorting order when we draw so it draws the polygons in different orders because everything is same Z, its technically legal. My guess at the moment.

**Comment by Opaque79** [comment]
My specs are:
CPU - AMD Ryzen 7 2700X
GPU - AMD Radeon RX 5700 XT

I could roll back to 2.12.1 and lose the PSX Uzi sound

**Comment by walkawayy** [comment]
So far the issue seems limited to AMD GPUs. From Discord, there was another report where an AMD Ryzen 7 3750H with Radeon Mobile Gfx had the issue. Switching to the Nvidia GeForce GTX 1650 dual GPU fixed it.

**Comment by Opaque79** [comment]
Switching GPU's isn't really practical so it seems my only option so far is to go back to 2.12.1 and lose the PSX Uzi sound.

It only affects text boxes, the rest of the game is fine btw

**Comment by walkawayy** [comment]
> Switching GPU's isn't really practical so it seems my only option so far is to go back to 2.12.1 and lose the PSX Uzi sound.
> 
> It only affects text boxes, the rest of the game is fine btw

Of course. That user was on a laptop with a dual GPU. Just trying to gather data to find out why it's broken. I agree with @oziphantom about a Z Buffer issue. But I don't think it's new driver related because T1M 2.12 works fine. @carlo-bramini do you think one of the changes you made could have caused this?

**Comment by Opaque79** [comment]
> > Switching GPU's isn't really practical so it seems my only option so far is to go back to 2.12.1 and lose the PSX Uzi sound.
> > It only affects text boxes, the rest of the game is fine btw
> 
> Of course. That user was on a laptop with a dual GPU. Just trying to gather data to find out why it's broken. I agree with @oziphantom about a Z Buffer issue. But I don't think it's new driver related because T1M 2.12 works fine. @carlo-bramini do you think one of the changes you made could have caused this?

Z-buffer makes sense but odd how it only affects the passport and textboxes

**Comment by lahm86** [comment]
This is an extract from the log file from another user facing this issue (Radgryd) - the third line references a compatibility profile, whereas on my own it lists my GPU name. Perhaps it's a clue into what could be causing it. 
```
../src/gfx/context.c 79 GFX_Context_Attach OpenGL vendor string:   ATI Technologies Inc.
../src/gfx/context.c 80 GFX_Context_Attach OpenGL renderer string: Radeon RX 580 Series
../src/gfx/context.c 81 GFX_Context_Attach OpenGL version string:  4.6.0 Compatibility Profile Context 23.2.2.230217
../src/gfx/context.c 85 GFX_Context_Attach Shading version string: 4.60
../src/gfx/context.c 53 GFX_Context_CheckExtensionSupport GL_ARB_explicit_attrib_location supported: yes
../src/gfx/context.c 53 GFX_Context_CheckExtensionSupport GL_EXT_gpu_shader4 supported: yes
```

**Comment by carlo-bramini** [comment]
Actually, changing this line:
https://github.com/LostArtefacts/Tomb1Main/blob/8df44b1dd6f5b66ac8a2a84678f94e8d52cf496c/src/specific/s_shell.c#L166
by changing 16 to 24 into `SDL_GL_DEPTH_SIZE` has fixed the problem on my side.

**Comment by walkawayy** [comment]
> 

Looks like the old setting before the change to SDL_GL_DEPTH_SIZE was `pfd.cDepthBits = 32;`.

**Comment by Opaque79** [comment]
How easy would this be to correct?

**Comment by Bi0Cell** [comment]
Same bug. My specs are:

AMD A12-9700P RADEON R7, 10 
COMPUTE CORES 4C+6G   2.50 GHz

**Comment by Opaque79** [comment]
It's worth mentioning that this affects some textures on Lara too, such as her waist showing through her belt

**Comment by lahm86** [comment]
@Opaque79, @Bi0Cell - here is a test build with the fix proposed by @carlo-bramini. If you get a chance, could you test this on your setups? Would be good to verify that it fixes the issue for all affected.
[Tomb1Main.zip](https://github.com/LostArtefacts/Tomb1Main/files/10930589/Tomb1Main.zip)

Do note that this is a test build, so please avoid distributing this file - if the problem is solved, it would be best to wait for the next official release containing the fix.

**Comment by Opaque79** [comment]
The new file has fixed the problem on my end, text boxes now display as they should

**Comment by rr-** [comment]
Fixed in 2.13.2
